http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java

```diff
@@ -116,23 +109,18 @@ private void addRequestConfig(final HttpRequestBase base,
 
 
     private HttpRequestBase createApacheRequest(HttpExecuteRequest request, URI uri) {
-        switch (request.httpRequest().method()) {
+        SdkHttpMethod method = request.httpRequest().method();
+        switch (method) {
             case HEAD:
-                return new HttpHead(uri);
             case GET:
-                return new HttpGet(uri);
             case DELETE:
-                return new HttpDelete(uri);
             case OPTIONS:
-                return new HttpOptions(uri);
```

Comment:
For `Options`, can we continue to use `HttpOptions` since it includes the `getAllowedMethods` method

```
return wrapEntity(request, new HttpOptions(uri));
```

Reply:
`getAllowedMethods` is not used by Apache HttpClient itself or by aws-sdk, and the Apache request objects aren't ever exposed to end users of the SDK. So I don't think there's any significant benefit to retaining it, but happy to rework things. Would however mean no entity on `OPTIONS` or having to reimplement all entity handling just on a subtype of `HttpOptions`

Reply:
Makes sense, we can keep as is